### DISCUSSION TITLE
Is it possible to use object pool to optimize performance?

### DISCUSSION THREAD

**Discussion by goldwind-ting** [tokio-rs/axum#1406] [Q&A] [answered]
What about using object pool to reuse some object, like `request`, `context`?

**Comment by davidpdrsn** [comment, marked as answer]
Nope

**Reply by goldwind-ting** [reply]
Could kindly share why not?

**Reply by davidpdrsn** [reply]
Honestly I'm not even sure what an object pool is. Haven't heard of another rust web framework, or even rust library, that uses it.

**Reply by goldwind-ting** [reply]
Thank you for your fast response. I found [fasthttp](https://github.com/valyala/fasthttp) use this.

**Reply by davidpdrsn** [reply]
Is that because it uses a GC? Does that still apply to Rust?

**Reply by goldwind-ting** [reply]
To some extent I think it is, but it also can reuse objects and cache allocated. please refer to the [benchmark](https://github.com/zslayton/lifeguard) at the bottom.
